src/codegen/codegen-expression.lisp

```diff
@@ -136,7 +136,7 @@
     (let ((match-expr (codegen-expression (node-while-let-expr expr) current-function env))
           (body-expr (codegen-expression (node-while-let-body expr) current-function env))
           (label (node-while-let-label expr))
-          (match-var (gensym "MATCH")))
+          (match-var (gentemp "MATCH")))
```

Comment:
Gensyms are not interned so can't be used as bindings in code that will survive a prin1/read cycle; however gentemp will litter the codegen package. Define a temporary package for these?

Reply:
This *should* be fine. `gentemp` will by default intern into `*package*` which should be bound to the package being compiled into.